### PR TITLE
Add Extension properties ordering

### PR DESCRIPTION
**Pull request by zayo**
I've noticed that detekt is treating extension properties as regular properties.
Opened issue on detekt: https://github.com/detekt/detekt/issues/3532
Lead to proposal: https://github.com/detekt/detekt/pull/3559

Yet that change would be unclear as this is not part of the spec at all. Thus proposing the following change.

### CODE REVIEW

**Review by BraisGabin — APPROVED** [review]

**Comment by BraisGabin** [comment]
This is not stale. I think that it should be merged or at least denied by a human telling why it should not be merged.

**Comment by p7nov** [comment]
Hey! We're currently testing some automation to help us handle PR from contributors in time. Sorry for not responding for so long. We'll review your suggestion in a couple of days.

**Comment by elizarov** [comment]
This addition contradicts the spirit of the Kotlin style guide which explicitly does not tries to mandate any order and says: " "Choose an order (either higher-level stuff first, or vice versa) and stick to it." It is up to specific projects and tools to decide upon any specific ordering requirements if they wish to enforce them.

**Comment by zayo** [comment]
Thats true, but guys from detekt told me, that this has unless it's in doc, they won't approve it.
The problem is, that extension properties are treated as regular properties, which I believe is wrong. Yet the doc does not states what the ext property is.
Any suggestion how to rephrase it to fit both needs?

**Comment by elizarov** [comment]
Kotlin style guide does not enforce the ordering for a reason. We have found that it is hard to come up with rock-solid ordering rules even for our own code, so we've bailed out of trying to solve it. 

It is up to detekt team to decide upon such rules if they wish to tackle this issue to have ordering enforced automatically.

**Comment by zayo** [comment]
I got your point. I don't require the doc to contain info about order.
I would like the doc to state what the extension property is. Is it property? Is it function? Is it neither one of them?

**Comment by elizarov** [comment]
I don't see how that kind of information is related to Kotlin's style guide.

**Comment by zayo** [comment]
You cannot say to not force order on properties unless you start calling them "properties". The doc says nothing about extensions. It doesn't even give a hint how they should be treated. I don't require to tell where they should be, but current state gives everyone chnace to mix them with methods(which they effectively are), or treating them like properties(which they look like). Which gives too much power and can be even harmful and error prone from my POV.
If that ends up with mixing the extension properties into regular properties, it's super confusing for the reader.

So I don't want to state how to order them, but how not to order them in a first place. To avoid code being error prone and confusing for the reader. Hope this is clear explanation why.

**Comment by elizarov** [comment]
This kind of decision is simply out of the scope of this particular document. It does not belong here.

**Comment by chao2zhang** [comment]
I believe this is within the scope of this document, particularly about clarifications of the existing doc `1. Property declarations and initializer blocks`. The proposal here is to clarify the `property` as non-extension properties because extension properties do get parsed as `KtProperty`.

On the other hand, I agree with you that it makes sense to not put any restrictions on extension properties in the official documentation.
